owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt

```diff
@@ -61,10 +62,17 @@ class MainFileListFragment : Fragment() {
     private fun initViews() {
         //Set RecyclerView and its adapter.
         fileListAdapter = FileListAdapter(context = requireContext())
+        fileListAdapter.onItemClick = { file ->
+            if (file.isFolder) {
+                mainFileListViewModel.listDirectory(file)
+                // TODO Manage animation listDirectoryWithAnimationDown
+            } else { /// Click on a file
+                // TODO Click on a file
+            }
+        }
```

Comment:
If you add the `onClick()` as parameter this could be changed to:
 `fileListAdapter = FileListAdapter(context = requireContext(), item -> )`